---
layout: workshop
root: .
venue: Harvard University
address: 356 Jefferson Labs, 17 Oxford Street, Cambridge, MA 02138  
country: United-States
humandate: August 19-20, 2015
humantime: 9:00 am - 5:00 pm
startdate: 2015-08-19
enddate: 2015-08-20
latlng: 
registration: open
host: ["Bob Freeman (Harvard FASRC)"]
instructor: ["Adam Freedman","Bob Freeman","Radhika Khetani","Meeta Mistry"
]
helper:  ["George Marnellos", "Mary Piper", "Tim Sackton"]
assistant: ["George Marnellos", "Mary Piper", "Tim Sackton"]
contact: robert_freeman@harvard.edu
raw: raw.github.com/datacarpentry/2015-08-19-Harvard/gh-pages
eventbrite: 17487821545
etherpad: https://harvardinformatics.etherpad.mozilla.org/2015-08-19-HU-FASRC
day1_pm: "Data processing at the command line"
day2_am: "Working with NGS data"
day2_pm: "Data analysis and visualization in R"
---
<!--
  HEADER

  Edit the values in the block above to be appropriate for your
  workshop.  Run 'tools/check' *before* committing to make sure that
  changes are good.
-->

<!--
  EVENTBRITE

  This block includes the Eventbrite registration widget if
  'eventbrite' has been set in the header.  You can delete it if you
  are not using Eventbrite, or leave it in, since it will not be
  displayed if the 'eventbrite' field in the header is not set.
-->
{% if page.eventbrite %}
<iframe
  src="https://www.eventbrite.com/tickets-external?eid={{page.eventbrite}}&ref=etckt"
  frameborder="0"
  width="100%"
  height="206px"
  scrolling="auto">
</iframe>
{% endif %}

<h2>General Information</h2>

<!--
  INTRODUCTION

  Edit the general explanatory paragraph below if you want to change
  the pitch.
-->
<p>
<a href=http://datacarpentry.org>Data Carpentry</a> workshops are for any researcher 
who has data they want to analyze, and no prior computational experience is required. 
This hands-on workshop teaches basic concepts, skills and tools for working more 
effectively with data. Less time, less pain.
</p>

<p><b>The focus of this workshop will be on working with genomics data
and data management and analysis for genomics research.</b> 
We will cover metadata organization in spreadsheets, connecting to 
and using a high-performance computing environment (Odyssey), the command line for 
sequence quality control and bioinformatics workflows,
and R for data analysis and visualization. 
We will not be teaching any particular bioinformatics tools, but the 
foundational skills that will allow you to conduct any analysis and
analyze the output of a genomics pipeline. 
</p>
<p>
Participants should bring their laptops and plan to participate actively. By the end of the workshop learners should be able to more effectively manage and analyze data and be able to apply the tools and approaches directly to their ongoing research.
</p>


<!--
<p>Data files for the workshop are available at <a href = "/2015-01-15-cornell/data/biology/">the following link</a></p>

<p>The etherpad for this workshop can be found <a href = "https://etherpad.mozilla.org/7XixtJSZJq">here</a></p>

<p>A static copy of the etherpad as of 5:30 on Friday after the workshop ended is <a href ="etherpad-snapshot-530-Friday.html">here</a></p>
-->
<p>The etherpad for this workshop can be found at <a href = "https://harvardinformatics.etherpad.mozilla.org/3">https://harvardinformatics.etherpad.mozilla.org/3</a></p>


<!--
  AUDIENCE

  Explain who your audience is.  (In particular, tell readers if the
  workshop is only open to people from a particular institution.
-->
<p>
  <strong>Who:</strong>
  The course is aimed at faculty, research staff, postdocs, graduate students, advanced undergraduates, and other researchers in any field. No prior computational experience is required, though we do expect familiarity with genomics concepts.
</p>

<!--
  LOCATION

  This block displays the address and links to maps showing directions
  if the latitude and longitude of the workshop have been set.  You
  can use http://itouchmap.com/latlong.html to find the lat/long of an
  address.
-->
{% if page.latlng %}
<p>
  <strong>Where:</strong>
  {{page.address}}.
  Get directions with
  <a href="//www.openstreetmap.org/?mlat={{page.latlng | replace:',','&mlon='}}&zoom=16">OpenStreetMap</a>
  or
  <a href="//maps.google.com/maps?q={{page.latlng}}">Google Maps</a>.
</p>
{% endif %}

<p>
  <strong>Where:</strong>
  {{page.address}}.
  Get directions with <a href="https://www.google.com/maps/place/17+Oxford+St,+Harvard+University,+Cambridge,+MA+02138/@42.3772635,-71.1166834,18.22z/data=!4m2!3m1!1s0x89e377416f2a5e07:0x27325c14d24b2507">Google Maps</a>.
</p>


<!--
  SPECIAL REQUIREMENTS

  Modify the block below if there are any special requirements.
-->
<p>
  <strong>Requirements:</strong> Participants must bring a laptop with
  a few specific software packages installed (listed
  <a href="#setup">below</a>). They are also required to abide by
  Data Carpentry's
  <a href="http://software-carpentry.org/conduct.html">Code of Conduct</a>.
</p>

<!--
  CONTACT EMAIL ADDRESS

  Display the contact email address set in the header.  If an address
  isn't set in the header, the Software Carpentry admin address is
  used.
-->
<p>
  <strong>Contact</strong>:
  Please mail
  {% if page.contact %}
    <a href='mailto:{{page.contact}}'>{{page.contact}}</a>
  {% else %}
    <a href='mailto:{{site.contact}}'>{{site.contact}}</a>
  {% endif %}
  for more information.
</p>

<hr/>

<!--
<p>
  <strong>Note on Cloud Computing</strong><br>
  The servers used in workshop are run in Amazon Web Services, courtesy of a grant from Amazon.
  If you want to run your instance of the server used for this workshop, launch 
  a t2.large instance with AMI ami-6516b30e, available under "Community AMIs" in the 
  Amazon EC2 Managemant Console.
  <br>
  <a href="http://aws.amazon.com/documentation/gettingstarted/">
    Getting started with AWS
  </a>
</p>
<hr>
-->

<!--
  SCHEDULE

  Show the workshop's schedule.  Edit the items and times in the table
  to match your plans.  You may also want to change 'Day 1' and 'Day
  2' to be actual dates or days of the week.
-->
<h2>Schedule</h2>

<div class="row">
  <div class="col-md-6">

    <h3>Day 1: 9AM - 5PM</h3>
    <table class="table table-striped">
      <tr> 
        <td colspan=2 style="background:gainsboro">Data Processing and Organization</td>
      </tr>
      
      <tr>
        <td>
          <a href="https://github.com/adamfreedman/datadescription-genomics/blob/master/lessons/01-intro-to-ecoli-evo-experient.md">Welcome, Logistics, and Lesson Introduction
        </td>
        <td>
            Bob/Adam
        </td>
      </tr>

      <tr>
        <td>
          <a href="https://github.com/adamfreedman/datadescription-genomics/blob/master/lessons/02-examining-sra-runtable.md">
            Getting started with data
          </a>
        </td>
        <td>
            Adam
        </td>
      </tr>

      <tr>
        <td>
            <a href="https://github.com/devbioinfoguy/shell-genomics/tree/gh-pages/lessons">
              Command line exploration of data (Unix shell part 1)
            </a>
        </td>
        <td>
            Bob
        </td
      </tr>

      <tr>
         <td>
          <a href="https://github.com/adamfreedman/organization-genomics/blob/gh-pages/lessons/01_intro_organization.md">
            Getting your project started
          </a>
        </td>
        <td>
            Adam
        </td>
      </tr>
      
      <tr>
        <td>
          
            <a href="https://github.com/devbioinfoguy/shell-genomics/tree/gh-pages/lessons">
              Command line exploration of data (Unix shell part 2)
            </a>
        </td>
        <td>
            Bob
        </td
      </tr>

      
      <tr>
        <td>
          
            
              Transitioning to HPC
            
        </td>
        <td>
            Bob
        </td
      </tr>


           <tr>
        <td>
          <a href="https://github.com/adamfreedman/knowyourdata-genomics/blob/gh-pages/lessons/01-know_your_data.md">
            Knowing your data
          </a>
        </td>
        <td>
          Adam
        </td>
      </tr>

    </table>
 
 
    <h3>Day 2: 8AM - 5AM</h3>
    <table class="table table-striped">
      <tr> 
        <td colspan=2 style="background:gainsboro">NGS data analysis</td>
      </tr>
      <tr> 
      <tr>
        <td>
          <a href="https://github.com/devbioinfoguy/wrangling-genomics/blob/gh-pages/lessons/00-readQC.md">
            Quality control of NGS data
          </a>
        </td>
        <td>
          Radhika
        </td>
      </tr>
       <tr>
        <td>
          <a href="https://github.com/devbioinfoguy/wrangling-genomics/blob/gh-pages/lessons/01-automating_a_workflow.md">
            Automating your workflow with a shell script
          </a>
        </td>
        <td>
          Radhika
        </td>
      </tr>
      <tr>
        <td>
          <a href="https://github.com/devbioinfoguy/wrangling-genomics/blob/gh-pages/lessons/02-variant-calling-workflow.md">
          Variant calling workflow and automation
          </a>
        </td>
        <td>
          Radhika
        </td>
      </tr>
      <tr>
        <td colspan=2 style="background:gainsboro">
          {{page.day2_pm}}
        </td>
      </tr>
      <tr>
        <td>
          <a href="http://tracykteal.github.io/R-genomics/">
            Data Exploration and visualization with R</a><br>
            Note: data available <a href="https://raw.githubusercontent.com/datacarpentry/R-genomics/master/Ecoli_genomes_with_SNPs_per_locus_tag.txt">here</a>
        </td>
        <td>
          Meeta
        </td>
      </tr>
            <tr>
        <td colspan=2 style="background:gainsboro">
          Wrap-up
        </td>
      </tr>
      <tr>
        <td>
          <a href="https://github.com/devbioinfoguy/cloud-genomics-HPC/blob/master/lessons/5.Data-rountripping.md">
            Moving your data</a>
        </td>
        <td>
          Meeta
        </td>
      </tr>
      <tr>
        <td>
          <a href="http://ucbpsych.qualtrics.com/SE/?SID=SV_9oaGksH1Q7KrMRn">
            Please complete the Post-Workshop Survey</a>
        </td>
        <td>
         
        </td>
      </tr>
    </table>
 </div>
</div>

<!--
  ETHERPAD

  Display the Etherpad for the workshop.  You can set this up in
  advance or on the first day; either way, make sure you push changes
  to GitHub after you have its URL.  To create an Etherpad, go to

      http://etherpad.mozilla.org/YYYY-MM-DD-site

  where 'YYYY-MM-DD-site' is the identifier for your workshop (e.g.,
  '2015-06-10-esu'.
-->
{% if page.etherpad %}
<p>
  <strong>Etherpad:</strong> <a href="{{page.etherpad}}">{{page.etherpad}}</a>.
  <br/>
  We will use this Etherpad for chatting, taking notes, and sharing URLs and bits of code.
</p>
{% endif %}

<hr/>

<!--
  SYLLABUS

  Show what topics will be covered.

  1. If your workshop is R rather than Python, remove the comment
     around that section and put a comment around the Python section.
  2. Some workshops will delete SQL.
  3. Please make sure the list of topics is synchronized with what you
     intend to teach.
  4. You may need to move the div's with class="col-md-6" around inside
     the div's with class="row" to balance the multi-column layout.

  This is one of the places where people frequently make mistakes, so
  please preview your site before committing, and make sure to run
  'tools/check' as well.
-->
<!--
<h2>Syllabus</h2>

<div class="row">
  <div class="col-md-6">
    <h3>The Unix Shell</h3>
    <ul>
      <li>Files and directories</li>
      <li>History and tab completion</li>
      <li>Pipes and redirection</li>
      <li>Looping over files</li>
      <li>Creating and running shell scripts</li>
      <li>Finding things</li>
      <li><a href="{{site.swc_lessons}}/ref/01-shell.html">Reference...</a></li>
    </ul>
  </div>
  <div class="col-md-6">
    <h3>Programming in Python</h3>
    <ul>
      <li>Using libraries</li>
      <li>Working with arrays</li>
      <li>Reading and plotting data</li>
      <li>Creating and using functions</li>
      <li>Loops and conditionals</li>
      <li>Defensive programming</li>
      <li>Using Python from the command line</li>
      <li><a href="{{site.swc_lessons}}/ref/03-python.html">Reference...</a></li>
    </ul>
  </div>

  <div class="col-md-6">
    <h3>Programming in R</h3>
    <ul>
      <li>Working with vectors and data frames</li>
      <li>Reading and plotting data</li>
      <li>Creating and using functions</li>
      <li>Loops and conditionals</li>
      <li>Using R from the command line</li>
      <li><a href="{{site.swc_lessons}}/ref/06-R.html">Reference...</a></li>
    </ul>
  </div>

</div>

<div class="row">
  <div class="col-md-6">
    <h3>Version Control with Git</h3>
    <ul>
      <li>Creating a repository</li>
      <li>Recording changes to files: <code>add</code>, <code>commit</code>, ...</li>
      <li>Viewing changes: <code>status</code>, <code>diff</code>, ...</li>
      <li>Ignoring files</li>
      <li>Working on the web: <code>clone</code>, <code>pull</code>, <code>push</code>, ...</li>
      <li>Resolving conflicts</li>
      <li>Open licenses</li>
      <li>Where to host work, and why</li>
      <li><a href="{{site.swc_lessons}}/ref/02-git.html">Reference...</a></li>
    </ul>
  </div>
  <div class="col-md-6">
    <h3>Managing Data with SQL</h3>
    <ul>
      <li>Reading and sorting data</li>
      <li>Filtering with <code>where</code></li>
      <li>Calculating new values on the fly</li>
      <li>Handling missing values</li>
      <li>Combining values using aggregation</li>
      <li>Combining information from multiple tables using <code>join</code></li>
      <li>Creating, modifying, and deleting data</li>
      <li>Programming with databases</li>
      <li><a href="{{site.swc_lessons}}/ref/04-sql.html">Reference...</a></li>
    </ul>
  </div>
</div>


<hr/>
-->

<!--
  SETUP

  Delete irrelevant sections from the setup instructions.  Each
  section is inside a 'div' without any classes to make the beginning
  and end easier to find.

  This is the other place where people frequently make mistakes, so
  please preview your site before committing, and make sure to run
  'tools/check' as well.
-->

<h2 id="setup">Setup</h2>

<p>
To participate in a Data Carpentry workshop, you will need working copies of the 
described software. Please make sure to install everything (or at least to download 
the installers) before the start of your workshop. Participants should bring and use 
their own laptops to insure the proper setup of tools for an efficient workflow 
once you leave the workshop.
<p>
Please follow these <a href=install.html>Setup Instructions</a>

<p>
  We maintain a list of common issues that occur during installation as a reference for instructors
  that may be useful on the
  <a href = "https://github.com/swcarpentry/workshop-template/wiki/Configuration-Problems-and-Solutions">Configuration Problems and Solutions wiki page</a>.
</p>

<h2>Acknowledgements &amp; Support</h2>
<!--
<div class="pull-right" style="max-width: 320px; padding-left: 6px;">
</div>
-->
<p>
Data Carpentry is supported by the <a href=http://http://www.moore.org/>Gordon and Betty Moore Foundation</a> and a partnership of several NSF-funded <a href="http://www.nsf.gov/dir/index.jsp?org=BIO" target="_blank">BIO</a> Centers (<a href="http://nescent.org" target="_blank">NESCent</a>, <a href="http://iplantcollaborative.org" target="_blank">iPlant</a>, <a href="http://idigbio.org" target="_blank">iDigBio</a>, <a href="http://beacon-center.org/" target="_blank">BEACON</a> and <a href="http://sesync.org" target="_blank">SESYNC</a>) and <a href="http://software-carpentry.org" target="_blank">Software Carpentry</a>. 
The structure and objectives of the curriculum as well as the teaching style are informed by <a href="http://software-carpentry.org" target="_blank">Software Carpentry</a>.
</p>
